utils/build_stig_control.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import argparse
+import json
+import os
+import sys
+
+import xml.etree.ElementTree as ET
+import yaml
+
+import ssg.build_yaml
+import ssg.environment
+import ssg.rules
+import ssg.yaml
+
+
+SSG_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+BUILD_OUTPUT = os.path.join(SSG_ROOT, "build", "stig_control.yml")
+RULES_JSON = os.path.join(SSG_ROOT, "build", "rule_dirs.json")
+BUILD_CONFIG = os.path.join(SSG_ROOT, "build", "build_config.yml")
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-r", "--root", type=str, action="store", default=SSG_ROOT,
+                        help="Path to SSG root directory (defaults to %s)" % SSG_ROOT)
+    parser.add_argument("-o", "--output", type=str, action="store", default=BUILD_OUTPUT,
+                        help="File to write yaml output to (defaults to build/stig_control.yml)")
+    parser.add_argument("-p", "--product", type=str, action="store", required=True,
+                        help="What product to get STIGs for")
+    parser.add_argument("-m", "--manual", type=str, action="store", required=True,
+                        help="Path to XML XCCDF manual file to use as the source of the STIGs")
+    parser.add_argument("-j", "--json", type=str, action="store", default=RULES_JSON,
+                        help="Path to the rules_dir.json (defaults to build/stig_control.json)")
+    parser.add_argument("-c", "--build-config-yaml", default=BUILD_CONFIG,
+                        help="YAML file with information about the build configuration. ")
+    parser.add_argument("-ref", "--reference", type=str, default="stigid",
+                        help="Reference system to check for, defaults to stigid")
+
+    return parser.parse_args()
+
+
+def handle_rule_yaml(args, rule_id, rule_dir, guide_dir):
+    rule_obj = {'id': rule_id, 'dir': rule_dir, 'guide': guide_dir}
+    rule_file = ssg.rules.get_rule_dir_yaml(rule_dir)
+
+    product_dir = os.path.join(args.root, "products", args.product)
+    product_yaml_path = os.path.join(product_dir, "product.yml")
```

Comment:
Not sure how long this takes to run, but as an optimization, `env_yaml` should be hoisted up and passed in once. Then it can be modified to include `rule_id` And such.

Reply:
Look up `local_env_yaml` in the code base :-)